templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,43 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+csr = STDIN.read
+
+settings_file    = File.exist?('/etc/puppetlabs/puppet/foreman.yaml') ? '/etc/puppetlabs/puppet/foreman.yaml' : '/etc/puppet/foreman.yaml'
+SETTINGS         = YAML.load_file(settings_file)
+protocol         = "<%= scope.lookupvar("foreman_proxy::puppetca_listen_on") %>"
+port             = "<% if scope.lookupvar("foreman_proxy::puppetca_listen_on") == "https" %><%= scope.lookupvar("foreman_proxy::ssl_port") -%>
+                    <%- else %><%= scope.lookupvar("foreman_proxy::http_port") -%>
+                    <%- end %>"
+# e.g. https://hostname.localdomain.com:8443/puppet/ca/autosign
+uri              = URI.parse("#{protocol}://<%= @fqdn %>:#{port}/puppet/ca/autosign")
+res              = Net::HTTP.new(uri.host, uri.port)
```

Comment:
https://ruby-doc.org/stdlib-2.0.0/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-How+to+use+Net-3A-3AHTTP suggests to use a block. https://ruby-doc.org/stdlib-2.0.0/libdoc/net/http/rdoc/Net/HTTP.html#class-Net::HTTP-label-HTTPS doesn't mention client certs though.

Reply:
I'm using the `start()` in a block, see line 33. But in order to get the certs working will need to create the object first and then start.